src/java/picard/analysis/CollectWgsMetrics.java

```diff
@@ -349,12 +358,18 @@ protected WgsMetrics generateWgsMetrics() {
         return new WgsMetrics();
     }
 
+    /**
+     * By design, if an INTERVALS file is provided, we want to count just those bases at the positions we care about,
+     * not across the entire read. Therefore, we call filter.getFilteredRecords() so that only the bases in the pileup
+     * at a given position are included in the calculations (with filter.getFilteredBases() we would be including other
+     * bases in the read too).
+     */
     protected long getBasesExcludedBy(final CountingFilter filter) {
-        return filter.getFilteredBases();
+        return (INTERVALS != null) ? filter.getFilteredRecords() : filter.getFilteredBases();
```

Comment:
this only works when "sampled sites" are a single base here and there. This needs to work when INTERVALS is complete contigs...and I think it doesn't. @eitanbanks could you help @meganshand figure out how to make it work...this might be a little more effort than we first thought.